src/index.ts

```diff
@@ -38,14 +65,14 @@ export function parseXY(text, options = {}) {
 
   let lines = text.split(/[\r\n]+/);
 
-  let matrix = [];
-  let info = [];
+  let matrix: number[][] = [];
+  let info: { position: number; value: string }[] = [];
   let position = 0;
-  for (let l = 0; l < lines.length; l++) {
-    let line = lines[l].trim();
+  lines.forEach((line) => {
+    line = line.trim();
     // we will consider only lines that contains only numbers
-    if (line.match(/[0-9]+/) && line.match(/^[0-9eE,;. \t+-]+$/)) {
-      let fields = line.split(/,[; \t]+|[; \t]+/);
+    if (/[0-9]+/.exec(line) && /^[0-9eE,;. \t+-]+$/.exec(line)) {
+      let fields: string[] = line.split(/,[; \t]+|[; \t]+/);
```

Comment:
I think you don't need to set the type explicitly here, typescript will infer it for you.

Reply:
also, `.test()` is better than `.exec()` if you only care about the boolean value

Reply:
- type deleted and
-  .test() used